lib/net/pool.js

```diff
@@ -3631,6 +3779,11 @@ class PoolOptions {
       this.bip37 = options.bip37;
     }
 
+    if (options.bip157 != null) {
+      assert(typeof options.bip157 === 'boolean');
+      this.bip157 = options.bip157;
+    }
```

Comment:
We need a check somewhere that the filter index exists if the user sets this option. It's tricky because the only module that is able to do that at all is `FullNode`, and this will be the first check of this kind in that module (because FullNode is what connects Indexer <-> Pool). See chaindb.verifyFlags() for some kind of reference.

We also need to expose the option `--bip157` in FullNode, and document that new option in the CHANGELOG.